### ISSUE TITLE
Snowflake PIVOT aliases unparsable

### ISSUE THREAD

**Issue by robert-norberg** [sqlfluff/sqlfluff#7244]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Snowflake supports naming pivot values directly in the `IN` clause within a `PIVOT`, which avoids having to use quoted identifiers (e.g. `SELECT "'2023_Q1'" AS q1_2023_sales`) or positional identifiers (e.g. `SELECT $1 AS q1_2023_sales`) in the `SELECT` and `WHERE` clauses. I think this is both cleaner and less fragile SQL, but this results in a `sqlfluff` parsing error.

```sql
SELECT
    empid AS employee_id,
    q1_2023_sales,
    q2_2023_sales,
    q3_2023_sales,
    q4_2023_sales
FROM quarterly_sales
PIVOT (
    SUM(amount) FOR quarter IN (
        '2023_Q1' AS q1_2023_sales,
        '2023_Q2' AS q2_2023_sales,
        '2023_Q3' AS q3_2023_sales,
        '2023_Q4' AS q4_2023_sales
    )
)
WHERE q1_2023_sales IS NOT NULL;
```

```bash
sqlfluff lint --dialect snowflake ./pivot_aliases.sql
```

```
== [pivot_aliases.sql] FAIL
L:   8 | P:   1 | AL01 | Implicit/explicit aliasing of table.
                       | [aliasing.table]
L:   8 | P:   1 | AL05 | Alias 'PIVOT' is never used in SELECT statement.
                       | [aliasing.unused]
L:   8 | P:   1 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   8 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   8 | P:   1 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:   8 | P:   6 | LT02 | Expected line break and no indent before '('.
                       | [layout.indent]
L:   8 | P:   7 |  PRS | Line 8, Position 7: Found unparsable section: '(\n
                       | SUM(amount) FOR quarter IN (\n     ...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

I end up having to disable parsing for the entire `PIVOT` clause to pass my `pre-commit` hook, which then results in a number of false positive rule violations, so I also have to tag the `PIVOT` line with `--noqa: ALL`.

```sql
SELECT
    empid AS employee_id,
    q1_2023_sales,
    q2_2023_sales,
    q3_2023_sales,
    q4_2023_sales
FROM quarterly_sales
-- noqa: disable=PRS
PIVOT ( -- noqa: ALL
    SUM(amount) FOR quarter IN (
        '2023_Q1' AS q1_2023_sales,
        '2023_Q2' AS q2_2023_sales,
        '2023_Q3' AS q3_2023_sales,
        '2023_Q4' AS q4_2023_sales
    )
) -- noqa: enable=PRS
WHERE q1_2023_sales IS NOT NULL;
```

### Expected Behaviour

I believe the `AS` keywords in the `PIVOT` clause should be parsed as `alias_operator` keywords the same as the `AS` in the `SELECT` clause.

### Observed Behaviour

```bash
sqlfluff parse --logger parser --ignore-local-config --dialect snowflake ./pivot_aliases.sql
```

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    column_reference:
[L:  2, P:  5]      |                        naked_identifier:                     'empid'
[L:  2, P: 10]      |                    whitespace:                               ' '
[L:  2, P: 11]      |                    alias_expression:
[L:  2, P: 11]      |                        [META] indent:
[L:  2, P: 11]      |                        alias_operator:
[L:  2, P: 11]      |                            keyword:                          'AS'
[L:  2, P: 13]      |                        whitespace:                           ' '
[L:  2, P: 14]      |                        naked_identifier:                     'employee_id'
[L:  2, P: 25]      |                        [META] dedent:
[L:  2, P: 25]      |                comma:                                        ','
[L:  2, P: 26]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                select_clause_element:
[L:  3, P:  5]      |                    column_reference:
[L:  3, P:  5]      |                        naked_identifier:                     'q1_2023_sales'
[L:  3, P: 18]      |                comma:                                        ','
[L:  3, P: 19]      |                newline:                                      '\n'
[L:  4, P:  1]      |                whitespace:                                   '    '
[L:  4, P:  5]      |                select_clause_element:
[L:  4, P:  5]      |                    column_reference:
[L:  4, P:  5]      |                        naked_identifier:                     'q2_2023_sales'
[L:  4, P: 18]      |                comma:                                        ','
[L:  4, P: 19]      |                newline:                                      '\n'
[L:  5, P:  1]      |                whitespace:                                   '    '
[L:  5, P:  5]      |                select_clause_element:
[L:  5, P:  5]      |                    column_reference:
[L:  5, P:  5]      |                        naked_identifier:                     'q3_2023_sales'
[L:  5, P: 18]      |                comma:                                        ','
[L:  5, P: 19]      |                newline:                                      '\n'
[L:  6, P:  1]      |                whitespace:                                   '    '
[L:  6, P:  5]      |                select_clause_element:
[L:  6, P:  5]      |                    column_reference:
[L:  6, P:  5]      |                        naked_identifier:                     'q4_2023_sales'
[L:  6, P: 18]      |                [META] dedent:
[L:  6, P: 18]      |            newline:                                          '\n'
[L:  7, P:  1]      |            from_clause:
[L:  7, P:  1]      |                keyword:                                      'FROM'
[L:  7, P:  5]      |                whitespace:                                   ' '
[L:  7, P:  6]      |                from_expression:
[L:  7, P:  6]      |                    [META] indent:
[L:  7, P:  6]      |                    from_expression_element:
[L:  7, P:  6]      |                        table_expression:
[L:  7, P:  6]      |                            table_reference:
[L:  7, P:  6]      |                                naked_identifier:             'quarterly_sales'
[L:  7, P: 21]      |                        newline:                              '\n'
[L:  8, P:  1]      |                        alias_expression:
[L:  8, P:  1]      |                            [META] indent:
[L:  8, P:  1]      |                            naked_identifier:                 'PIVOT'
[L:  8, P:  6]      |                            [META] dedent:
[L:  8, P:  6]      |                    [META] dedent:
[L:  8, P:  6]      |            whitespace:                                       ' '
[L:  8, P:  7]      |            unparsable:                                       !! Expected: 'Nothing here.'
[L:  8, P:  7]      |                start_bracket:                                '('
[L:  8, P:  8]      |                newline:                                      '\n'
[L:  9, P:  1]      |                whitespace:                                   '    '
[L:  9, P:  5]      |                word:                                         'SUM'
[L:  9, P:  8]      |                start_bracket:                                '('
[L:  9, P:  9]      |                word:                                         'amount'
[L:  9, P: 15]      |                end_bracket:                                  ')'
[L:  9, P: 16]      |                whitespace:                                   ' '
[L:  9, P: 17]      |                word:                                         'FOR'
[L:  9, P: 20]      |                whitespace:                                   ' '
[L:  9, P: 21]      |                word:                                         'quarter'
[L:  9, P: 28]      |                whitespace:                                   ' '
[L:  9, P: 29]      |                word:                                         'IN'
[L:  9, P: 31]      |                whitespace:                                   ' '
[L:  9, P: 32]      |                start_bracket:                                '('
[L:  9, P: 33]      |                newline:                                      '\n'
[L: 10, P:  1]      |                whitespace:                                   '        '
[L: 10, P:  9]      |                single_quote:                                 "'2023_Q1'"
[L: 10, P: 18]      |                whitespace:                                   ' '
[L: 10, P: 19]      |                word:                                         'AS'
[L: 10, P: 21]      |                whitespace:                                   ' '
[L: 10, P: 22]      |                word:                                         'q1_2023_sales'
[L: 10, P: 35]      |                comma:                                        ','
[L: 10, P: 36]      |                newline:                                      '\n'
[L: 11, P:  1]      |                whitespace:                                   '        '
[L: 11, P:  9]      |                single_quote:                                 "'2023_Q2'"
[L: 11, P: 18]      |                whitespace:                                   ' '
[L: 11, P: 19]      |                word:                                         'AS'
[L: 11, P: 21]      |                whitespace:                                   ' '
[L: 11, P: 22]      |                word:                                         'q2_2023_sales'
[L: 11, P: 35]      |                comma:                                        ','
[L: 11, P: 36]      |                newline:                                      '\n'
[L: 12, P:  1]      |                whitespace:                                   '        '
[L: 12, P:  9]      |                single_quote:                                 "'2023_Q3'"
[L: 12, P: 18]      |                whitespace:                                   ' '
[L: 12, P: 19]      |                word:                                         'AS'
[L: 12, P: 21]      |                whitespace:                                   ' '
[L: 12, P: 22]      |                word:                                         'q3_2023_sales'
[L: 12, P: 35]      |                comma:                                        ','
[L: 12, P: 36]      |                newline:                                      '\n'
[L: 13, P:  1]      |                whitespace:                                   '        '
[L: 13, P:  9]      |                single_quote:                                 "'2023_Q4'"
[L: 13, P: 18]      |                whitespace:                                   ' '
[L: 13, P: 19]      |                word:                                         'AS'
[L: 13, P: 21]      |                whitespace:                                   ' '
[L: 13, P: 22]      |                word:                                         'q4_2023_sales'
[L: 13, P: 35]      |                newline:                                      '\n'
[L: 14, P:  1]      |                whitespace:                                   '    '
[L: 14, P:  5]      |                end_bracket:                                  ')'
[L: 14, P:  6]      |                newline:                                      '\n'
[L: 15, P:  1]      |                end_bracket:                                  ')'
[L: 15, P:  2]      |                newline:                                      '\n'
[L: 16, P:  1]      |                word:                                         'WHERE'
[L: 16, P:  6]      |                whitespace:                                   ' '
[L: 16, P:  7]      |                word:                                         'q1_2023_sales'
[L: 16, P: 20]      |                whitespace:                                   ' '
[L: 16, P: 21]      |                word:                                         'IS'
[L: 16, P: 23]      |                whitespace:                                   ' '
[L: 16, P: 24]      |                word:                                         'NOT'
[L: 16, P: 27]      |                whitespace:                                   ' '
[L: 16, P: 28]      |                word:                                         'NULL'
[L: 16, P: 32]      |    statement_terminator:                                     ';'
[L: 16, P: 33]      |    newline:                                                  '\n'
[L: 17, P:  1]      |    [META] end_of_file:
```

### How to reproduce

Ran this to create an example table:

```sql
CREATE OR REPLACE TEMPORARY TABLE quarterly_sales (
    empid INT,
    amount INT,
    quarter TEXT
)
AS
SELECT *
FROM
    VALUES
    (1, 10000, '2023_Q1'),
    (1, 400, '2023_Q1'),
    (2, 4500, '2023_Q1'),
    (2, 35000, '2023_Q1'),
    (1, 5000, '2023_Q2'),
    (1, 3000, '2023_Q2'),
    (2, 200, '2023_Q2'),
    (2, 90500, '2023_Q2'),
    (1, 6000, '2023_Q3'),
    (1, 5000, '2023_Q3'),
    (2, 2500, '2023_Q3'),
    (2, 9500, '2023_Q3'),
    (3, 2700, '2023_Q3'),
    (1, 8000, '2023_Q4'),
    (1, 10000, '2023_Q4'),
    (2, 800, '2023_Q4'),
    (2, 4500, '2023_Q4'),
    (3, 2700, '2023_Q4'),
    (3, 16000, '2023_Q4'),
    (3, 10200, '2023_Q4');
```

Put this in its own file (`pivot_aliases.sql`) for parsing.

```sql
SELECT
    empid AS employeeid,
    q1_2023_sales,
    q2_2023_sales,
    q3_2023_sales,
    q4_2023_sales
FROM quarterly_sales
PIVOT (
    SUM(amount) FOR quarter IN (
        '2023_Q1' AS q1_2023_sales,
        '2023_Q2' AS q2_2023_sales,
        '2023_Q3' AS q3_2023_sales,
        '2023_Q4' AS q4_2023_sales
    )
)
WHERE q1_2023_sales IS NOT NULL;

```

Parse it using CLI:

```bash
sqlfluff parse --logger parser --ignore-local-config --dialect snowflake ./pivot_aliases.sql
```

### Dialect

snowflake

### Version

```bash
sqlfluff --version
```

```
sqlfluff, version 3.5.0
```

```bash
python --version
```

```
Python 3.13.3
```

### Configuration

N/A, ran `sqlfluff parse` with `--ignore-local-config` to avoid config-specific issues.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
